### DISCUSSION TITLE
yo, im trying to add my own song, but it dosent work.

### DISCUSSION THREAD

**Discussion by imcomplingstuff** [JordanSantiagoYT/FNF-JS-Engine#644] [Q&A] [answered]
i go to the chart editor and put the name of the song's folder, but then it just dosent load anything and, for example, if i was on the song guns, it plays the same song even after putting the "reload audio" option. Can anybody help?

**Comment by JordanSantiagoYT** [comment]
that's because you can't add custom songs through the "assets/songs" folder. you have to add it into "mods/songs"

**Reply by imcomplingstuff** [reply]
thats...
Its juts literally what i did most of the time i tried to add a song to it

**Reply by JordanSantiagoYT** [reply]
> thats... Its juts literally what i did most of the time i tried to add a song to it

then any of the following could have happened:
- your song's name has spaces, and you didnt replace the spaces with dashes on the file path
- your inst/voices files aren't named "Inst" or "Voices" specifically, or they aren't in OGG format
- you have the "Load Songs" optimization option turned off

**Comment by imcomplingstuff** [comment]
2.
![image](https://github.com/user-attachments/assets/b7ab078c-b7ba-4efe-8f88-de2f78e7d936)

3.
![image](https://github.com/user-attachments/assets/325df322-b286-489d-8ae1-e0f2d6a86da3)

1.
![image](https://github.com/user-attachments/assets/a6152da7-0a3b-418c-ace6-3fcd998d9bcc)

**Reply by imcomplingstuff** [reply]
and no, it dosent have a "Voices.ogg" file.

**Reply by imcomplingstuff** [reply]
https://www.mediafire.com/file/ni5d2j3jj9udhp4/Showing-it-off-JSEngine.mp4/file
Here's a video about what happens, btw

**Reply by JordanSantiagoYT** [reply]
"mods/thebambi"

if your song isn't in the global "mods" folder, the game won't find your song unless the song in the chart editor is a song in the "thebambi" mod. because you are trying to do it from a base song, the game is trying to find "mods/lunar/Inst.ogg" which doesn't exist

**Reply by imcomplingstuff** [reply]
oh, but then what can i use for like, going into the song?
what can i do for going into it?

**Reply by imcomplingstuff** [reply]
tried doing one that thought that maybe worked
![image](https://github.com/user-attachments/assets/3f1f19ce-6e97-4aa3-be2b-3024f7d62957)
it crashed.
tried porting from my other mod on psych engine, a specific song, but it crashes.

(if you need this, here you go)
Uncaught Error: Invalid char 93 at position 73
C:\hostedtoolcache\windows\haxe\4.3.4\x64\std/haxe/format/JsonParser.invalidChar() [line 318]
C:\hostedtoolcache\windows\haxe\4.3.4\x64\std/haxe/format/JsonParser.parseRec() [line 110]
C:\hostedtoolcache\windows\haxe\4.3.4\x64\std/haxe/format/JsonParser.parseRec() [line 90]
C:\hostedtoolcache\windows\haxe\4.3.4\x64\std/haxe/format/JsonParser.doParse() [line 57]
WeekData.getWeekFile() [line 224]
WeekData.addWeek() [line 192]
WeekData.reloadWeekFiles() [line 180]
FreeplayState.create() [line 83]
flixel/FlxGame.switchState() [line 639]
flixel/FlxGame.update() [line 726]
flixel/FlxGame.step() [line 679]
flixel/FlxGame.onEnterFrame() [line 558]
openfl/events/EventDispatcher.__dispatchEvent() [line 426]
openfl/display/DisplayObject.__dispatch() [line 1398]

**Reply by JordanSantiagoYT** [reply]
that error means a week json you added is formatted incorrectly.

send me it and i'll fix it for you

**Reply by imcomplingstuff** [reply]
Alright, here you go
[week_songs.json](https://github.com/user-attachments/files/17319373/week_songs.json)

**Reply by JordanSantiagoYT** [reply]
fixed week json.

the issue was that you put a comma after the first and only song, even though there are no additional songs after that. the last song in a week shouldn't have a comma
[week_songs.json](https://github.com/user-attachments/files/17320002/week_songs.json)




what i mean:

broken 
![image](https://github.com/user-attachments/assets/816ca851-12b4-42f1-9e47-b2f724fde836)

fixed 
![image](https://github.com/user-attachments/assets/7f5b8ae2-e655-468c-8ba7-9e1896d6951b)

**Reply by JordanSantiagoYT** [reply, marked as answer]
[week_songs.json](https://github.com/user-attachments/files/17320008/week_songs.json)
re-fixed, you also forgot to put a name for the song

**Reply by imcomplingstuff** [reply]
thanks man, you actually helped me a lot by this single comment, my brain actually just started working, lol
tho i find it quite specific that, if a comma is there, it just dosent allow it to work AT ALL.

**Reply by JordanSantiagoYT** [reply]
> thanks man, you actually helped me a lot by this single comment, my brain actually just started working, lol tho i find it quite specific that, if a comma is there, it just dosent allow it to work AT ALL.

no worries man!

also, you can put more than 1 song in a week if you add a comma, then use the same template. for example:

```
	"songs": [
		[
			"Lunar",
			"cat",
			[
				146,
				113,
				253
			]
		],
		[
			"Song 2",
			"cat",
			[
				146,
				113,
				253
			]
		]
	],
